crates/cheatcodes/src/inspector.rs

```diff
@@ -982,9 +986,18 @@ impl Cheatcodes {
                             paymaster: paymaster_data.address.to_h160(),
                             paymaster_input: paymaster_data.input.to_vec(),
                         });
-                    if let Some(factory_deps) = zk_tx {
-                        let mut batched =
-                            foundry_zksync_core::vm::batch_factory_dependencies(factory_deps);
+                    if let Some(mut factory_deps) = zk_tx {
+                        for factory_deps_contract in &self.contracts_as_factory_deps {
+                            let bytecode =
+                                crate::fs::get_artifact_code(self, factory_deps_contract, false)
+                                    .unwrap_or_else(|_| panic!("Failed to get bytecode for factory deps contract {factory_deps_contract}"));
+                            info!("Pushing {:?} factory deps bytecode", factory_deps_contract);
+                            factory_deps.push(bytecode.to_vec());
+                        }
+                        self.contracts_as_factory_deps.clear();
```

Comment:
I wouldn't clear it for broadcast, as it happens before the actual call/create is ran, which would result in this latter process not having the required data (same reasoning as the `paymaster_params` above)